docs/dev/new_check_howto.md

```diff
@@ -269,58 +269,57 @@ The check is almost done. Let's add the final touches by adding the integration
+- **`README.md`**: This contains the documentation for your Check, how to set it up, which data it collects, etc…
+- **`spec.yaml`**: This is used to generate `conf.yaml.example` using the `ddev` tooling (see the "Configuration template" tab below). [See the configuration specification documentation to learn more.][16]
+- **`conf.yaml.example`**: This contains default (or example) configuration options for your Agent Check. Do not edit this file by hand—it is generated from the contents of `spec.yaml`. [See the configuration file reference documentation to learn its logic.][10]
+- **`manifest.json`**: This contains the metadata for your Agent Check such as the title, categories, etc… [See the manifest reference documentation to learn more.][11]
 - **`metadata.csv`**: This contains the list of all metrics collected by your Agent Check. [See the metrics metadata reference documentation to learn more.][12]
-- **`service_check.json`**: This contains the list of all Service Checks collected by your Agent check. [See the Service Check reference documentation to learn more.][13]
+- **`service_check.json`**: This contains the list of all Service Checks collected by your Agent Check. [See the Service Check reference documentation to learn more.][13]
 
 For this example, those files would have the following form:
 
 {{< tabs >}}
-{{% tab "Configuration file" %}}
+{{% tab "Configuration template" %}}
 
-The `awesome/datadog_checks/awesome/data/conf.yaml.example` for the Awesome service check:
+The `awesome/assets/configuration/spec.yaml` used to generate `awesome/datadog_checks/awesome/data/conf.yaml.example`:
 
 ```yaml
-init_config:
-  ## Block comment in the init_config
-  ## part.
-
-## Block comment outside
-## of the init_config part.
-
-instances:
-  ## @param url - string - required
-  ## The URL you want to check
-  ## (Note the indentation with the hyphen)
-  #
-  - url: http://example.org
-
-    ## @param search_string - string - required
-    ## The string to search for
-    #
-    search_string: "Example Domain"
-
-    ## @param user - object - optional
-    ## The user should map to the structure
-    ## {'name': ['<FIRST_NAME>', '<LAST_NAME>'], 'username': <USERNAME>, 'password': <PASSWORD>}
-    #
-    # user:
-    #   name:
-    #     - <FIRST_NAME>
-    #     - <LAST_NAME>
-    #   username: <USERNAME>
-    #   password: <PASSWORD>
-
-    ## @param options - object - required
-    ## Optional flags you can set
-    #
+name: awesome
+files:
+- name: awesome.yaml
+  options:
+  - template: init_config
+    options:
+    - template: init_config/default
+  - template: instances
     options:
-      ## @param follow_redirects - boolean - optional - default: false
-      ## Set to true to follow 301 Redirect
-      #
-      # follow_redirects: false
+    - name: url
+      required: true
+      description: "The URL to check."
```

Comment:
Let's remove quotes for easier viewing:

```suggestion
      description: The URL to check.
```

Reply:
Is this a style choice or a technical requirement? I'm ok with removing the quotes—I just want to know why. :)

Reply:
Style

> Let's remove quotes for easier viewing